### PR TITLE
Add repository layer tests for ShortUrl and ClickTracking

### PR DESCRIPTION
**Pull request by sadabazhar**
## Overview

This PR introduces comprehensive test coverage for the persistence layer, focusing on ShortUrlRepository and ClickTrackingRepository.

The goal is to validate correctness, constraints, and edge-case handling at the database interaction level, ensuring reliability before integrating higher-level services.

---

### 1.  ShortUrlRepository Tests:
✅ Sequence generation validation:
- Ensures uniqueness
- Ensures monotonic increase

✅ Query validation:

- findByShortCode
- existsByShortCode

✅ Constraint testing:

- Unique constraint on shortCode

✅ Expiry logic:

- Validates filtering of expired vs active URLs

### 2. ClickTrackingRepository Tests
✅ Batch increment logic:

- Single shortCode update
- Multiple shortCodes in one batch

✅ Failure scenarios:

- Invalid shortCode handling
- Exception propagation

⚠️ Observed behavior:

- Partial updates occur when batch fails (no rollback)

✅ Scalability test:

- Large batch processing (600+ records)

--- 

### Observations
⚠️ Partial Batch Update Issue

Test:

`shouldRollbackEntireBatch_whenAnyShortCodeIsInvalid`

Current Behavior:

Batch fails when one key is invalid
BUT already processed updates are not rolled back

### Testing

- Uses @DataJpaTest for lightweight JPA testing
- Uses TestEntityManager for controlled DB state
- Avoids in-memory DB replacement to match real DB behavior
- Covers both:
    - happy paths
    - failure scenarios

## Summary by CodeRabbit

* **Tests**
  * Added integration tests for click-tracking: single and multi-entry increments, invalid-short-code error handling, batch rollback behavior, and a high-volume batch scenario.
  * Added repository tests for short URL operations: sequence ID generation and ordering, retrieval and existence checks, duplicate short-code constraint, and counting only non-expired records.

### CODE REVIEW

File: src/test/java/com/rapidlink/repository/ClickTrackingRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        assertThrows(Exception.class,
+                () -> repository.batchIncrement(counts));
+
+        entityManager.clear();
+
+        ShortUrl after = shortUrlRepository.findByShortCode("valid123").get();
+
+        // Partial update DID happen
+        assertEquals(5L, after.getClickCount());
+    }
```
[the remark this answers is not in the document]
Comment (sadabazhar) [reply src/test/java/com/rapidlink/repository/ClickTrackingRepositoryTest.java:128]: I created an issue for it, I will fix it later

Issue: https://github.com/sadabazhar/rapidlink/issues/33#issue-4355635491

File: src/test/java/com/rapidlink/repository/ShortUrlRepositoryTest.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+        entityManager.persist(active);
+        entityManager.persist(future);
+        entityManager.persist(expired);
+        entityManager.flush();
+
+        long count = repository.countByExpiresAtAfterOrExpiresAtIsNull(now);
+
+        assertEquals(2, count); // active + future
+    }
```
[the remark this answers is not in the document]
Comment (sadabazhar) [reply src/test/java/com/rapidlink/repository/ShortUrlRepositoryTest.java:146]: Accept
